src/Metadata/Driver/YamlDriver.php

```diff
@@ -15,12 +15,16 @@
 use JMS\Serializer\Type\Parser;
 use JMS\Serializer\Type\ParserInterface;
 use Metadata\ClassMetadata as BaseClassMetadata;
-use Metadata\Driver\AbstractFileDriver;
+use Metadata\Driver\AdvancedDriverInterface;
+use Metadata\Driver\AdvancedFileLocatorInterface;
+use Metadata\Driver\FileLocator;
 use Metadata\Driver\FileLocatorInterface;
 use Metadata\MethodMetadata;
+use ReflectionClass;
+use RuntimeException;
 use Symfony\Component\Yaml\Yaml;
 
-class YamlDriver extends AbstractFileDriver
+class YamlDriver implements AdvancedDriverInterface
```

Comment:
Does it make sense just for compatibility to still extend `AbstractFileDriver`?

Reply:
All of it public and protected contracts were kept. Only `instance of AbstractFileDriver` should break. if it's not enougth we can keep extension with everything overrided